src/Hazelcast.Net/Sql/SqlService.cs

```diff
@@ -13,25 +13,36 @@
 // limitations under the License.
 
 using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Hazelcast.Clustering;
 using Hazelcast.Core;
+using Hazelcast.Messaging;
+using Hazelcast.Networking;
 using Hazelcast.Protocol.Codecs;
 using Hazelcast.Serialization;
+using Microsoft.Extensions.Logging;
 
 namespace Hazelcast.Sql
 {
     internal class SqlService : ISqlService
     {
         private readonly Cluster _cluster;
         private readonly SerializationService _serializationService;
+        // internal for tests only
+        internal readonly ReadOptimizedLruCache<string, int> _queryPartitionArgumentCache;
+        private readonly ILogger<SqlService> _logger;
+        private readonly HazelcastOptions _options;
 
-        internal SqlService(Cluster cluster, SerializationService serializationService)
+        internal SqlService(HazelcastOptions options, Cluster cluster, SerializationService serializationService, ILoggerFactory loggerFactory)
```

Comment:
i'd like to avoid passing the entire `HazelcastOptions` to the SQL service. looks like it only requires a `SqlOptions` instance + a flag to tell whether to do SQL partitioning or not. could be an internal `SqlOptions.PartitionEnabled` or something, that would be set by the client when creating the service. that way we don't leak any network stuff into the SQL service.

Reply:
Sure we can separate options into required fragments but I couldn't understand the concern here. What does it mean we don't leak network stuff since everything is internal in the context?

Reply:
Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/798/commits/a91d960491af90459c71f0821c8ccfdd5e85d119